yoti_python_sdk/anchor.py

```diff
@@ -1,141 +1,188 @@
-
-                                if anchor_type != "Unknown":
-                                    parsed_anchors = Anchor.get_values_from_extensions(anc, anchor_type, extensions,
-                                                                                       crypto_cert, parsed_anchors)
-                    except Exception as exc:
-                        if logging.getLogger().propagate:
-                            logging.warning('Error parsing anchor certificate extension, exception: {0} - {1}'.format(
-                                type(exc).__name__, exc))
-                        continue
-
-        return parsed_anchors
-
-    @staticmethod
-    def get_values_from_extensions(anc, anchor_type, extensions, crypto_cert, parsed_anchors):
-        if hasattr(extensions, 'value'):
-            extension_value = extensions.value
-            if hasattr(extension_value, 'value'):
-                parsed_anchors.append(Anchor(
-                    anchor_type,
-                    anc.sub_type,
-                    Anchor.decode_asn1_value(extension_value.value),
-                    anc.signed_time_stamp,
-                    crypto_cert))
-
-        return parsed_anchors
-
-    @staticmethod
-    def decode_asn1_value(value_to_decode):
-        extension_value_asn1 = value_to_decode
-        decoder = asn1.Decoder()
-
-        decoder.start(extension_value_asn1)
-        tag, once_decoded_value = decoder.read()
-
-        decoder.start(once_decoded_value)
-        tag, twice_decoded_value = decoder.read()
-
-        utf8_value = twice_decoded_value.decode('utf-8')
-        return utf8_value
-
-    @property
-    def anchor_type(self):
-        return self.__anchor_type
-
-    @property
-    def value(self):
-        return self.__value
-
-    @property
-    def sub_type(self):
-        return self.__sub_type
-
-    @property
-    def signed_timestamp(self):
-        if self.__signed_timestamp is None:
-            return None
-
-        signed_timestamp_object = compubapi.SignedTimestamp()
-        signed_timestamp_object.MergeFromString(self.__signed_timestamp)
-
-        try:
-            signed_timestamp_parsed = datetime.datetime.fromtimestamp(
-                signed_timestamp_object.timestamp / float(1000000))
-        except OSError:
-            print("Unable to parse timestamp from integer: '{0}'".format(signed_timestamp_object.timestamp))
-            return ""
-
-        return signed_timestamp_parsed
-
-    @property
-    def origin_server_certs(self):
-        return self.__origin_server_certs
+import datetime
+import logging
+
+import OpenSSL
+import asn1
+import yoti_python_sdk.protobuf.common_public_api.SignedTimestamp_pb2 as compubapi
+from OpenSSL import crypto
+
+from yoti_python_sdk import config
+
+UNKNOWN_EXTENSION = ""
+SOURCE_EXTENSION = "1.3.6.1.4.1.47127.1.1.1"
+VERIFIER_EXTENSION = "1.3.6.1.4.1.47127.1.1.2"
+
+UNKNOWN_ANCHOR_TYPE = "Unknown"
+UNKNOWN_ANCHOR_SUB_TYPE = "TEST UNKNOWN SUB TYPE"
```

Comment:
This should only be in the test (this is the sub type of the sample unknown anchor). The sub type should be available for all anchors.